### ISSUE TITLE
Modernize Quickfix List

### ISSUE THREAD

**Issue by badosu** [bryphe/oni#730]
This is one of the prospects that excited me for a long time since neovim was released.

Currently we use neovim's own quickfix list (the lower window):

![screenshot from 2017-09-27 22-25-15](https://user-images.githubusercontent.com/347552/30945166-ec1d2a1c-a3d2-11e7-9069-e503322693db.png)

We can leverage the UI possibilities provided by electron to:

- Make the quickfix window a proper window, separated from electron window or integrated, much like a console would.
- Improve the display of each quickfix entry (filename, column and text excerpt). E.g. highlighting the found part and truncating the content of the rest of the line to fit it into a sane layout.
- Improve the capabilities of the quickfix window to enable easier iteration through results.

**Comment by bryphe** [comment]
> This is one of the prospects that excited me for a long time since neovim was released.

Definitely! This is a terrific case to 'break free' of the terminal limitations and build something nice. It's somewhat related to what I was exploring in #622 - which is the ability to have 'custom renderers' on top of a Neovim surface. That then enables us to do all kinds of interesting UI on top of Neovim.... while still allowing keyboard navigation and behavior as you'd expect from Neovim.
